internal/go-prompt/document_test.go

```diff
@@ -0,0 +1,78 @@
+package prompt
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDocument_FindPreviousWordStart(t *testing.T) {
+	testCases := []struct {
+		doc         Document
+		charsToLeft int
+	}{
+		{
+			doc: Document{
+				Text:           "",
+				cursorPosition: 0,
+			},
+			charsToLeft: 0,
+		},
+		{
+			doc: Document{
+				Text:           "abc",
+				cursorPosition: 0,
+			},
+			charsToLeft: 0,
+		},
+		{
+			doc: Document{
+				Text:           "abc",
+				cursorPosition: 1,
+			},
+			charsToLeft: 1,
+		},
+		{
+			doc: Document{
+				Text:           "abc",
+				cursorPosition: 2,
+			},
+			charsToLeft: 2,
+		},
+		{
+			doc: Document{
+				Text:           "abc ",
+				cursorPosition: 3,
+			},
+			charsToLeft: 3,
+		},
+		{
+			doc: Document{
+				Text:           "abc  ",
+				cursorPosition: 4,
+			},
+			charsToLeft: 4,
+		},
+		{
+			doc: Document{
+				Text:           "abc x",
+				cursorPosition: 4,
+			},
+			charsToLeft: 4,
+		},
+		{
+			doc: Document{
+				Text:           "abc xw",
+				cursorPosition: 5,
+			},
+			charsToLeft: 1,
+		},
```

Comment:
```
doc: Document{
Text:       "    ",
    cursorPosition: 1,
},
charsToLeft: 0,
```
I'm not sure that it is the correct behavior or not but the test above returns one as expected even there is no word before the given cursor position. Shouldn't have it returned zero ?

Reply:
Since it finds the previous word starting position through current index of the curser according to whitespaces. It is clear that when we type sequential spaces then press `ctrl + left arrow`, it should return back and jump over all contiguous spaces to the beginning the given literal. I'm resolving.